src/cpp/qos/QosPolicies.cpp

```diff
@@ -39,6 +39,56 @@ uint32_t QosPolicy::get_cdr_serialized_size(
     return 2 + 2 + 4 + data_size;
 }
 
+bool QosPolicy::serialize_generic_data(
+        rtps::CDRMessage_t* msg,
+        uint16_t pid,
+        const std::vector<rtps::octet>& data)
+{
+    bool valid = CDRMessage::addUInt16(msg, pid);
+    uint16_t siz = static_cast<uint16_t>(data.size());
+    siz = (siz + 3) & ~3;
```

Comment:
Should we be aligning the size value? Shouldn't we be sending the real size of the data and be responsibility of the reader to skip the alignment before progressing the next submessage?

Reply:
True!

Reply:
The parameter length should be multiple of 4, according to the RTPS standard